docs/gpt.rst

```diff
@@ -0,0 +1,67 @@
+gpt
+=======
+
+.. dfhack-tool::
+    :summary: AI-generated written content!
+    :tags: fort gameplay
+
+Enables a UI for submitting knowledge item descriptions to OpenAI for generating
+poetry, star charts, and excerpts from longer works such as biographies, dictionaries,
+treatises on technological evolution, comparative biographies, cultural histories,
+autobiographies, cultural comparisons, essays, guides, manuals, and more.
+
+``enable gpt``
+=======
+Enables the plugin. The overlay will be shown when a knowledge item or unit view sheet is open.
+
+``disable gpt``
+=======
+Disables the plugin.
+
+Setup:
+
+1. Register for an OpenAI API account. It must be a paid or active trial account.
+2. Generate an API token for your account.
+3. Save your OpenAI API token to a file at the root of your DF directory, `oaak.txt`.
+4. Install python. We used version 3.11 installed from the Microsoft Store.
+5. Install python dependencies Flask and OpenAI: `pip install Flask` and `pip install OpenAI`.
+6. Start the local helper python app: cd into dfhack/scripts directory & run `python gptserver.py`.
```

Comment:
I see two options here. if we deploy this with DFHack, then this python script should probably go in the DFHack/dfhack repo and get installed somewhere more appropriate (like maybe the DF root dir)

if this script gets distributed as a mod (see https://docs.dfhack.org/en/stable/docs/guides/modding-guide.html#the-structure-of-a-mod), then the python code can get included in the mod files